inttest/capi-docker/capi_docker_test.go

```diff
@@ -0,0 +1,203 @@
+*/
+
+package capidocker
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/k0sproject/k0smotron/inttest/util"
+	"github.com/stretchr/testify/suite"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/k0sproject/k0s/inttest/common"
+	k0stestutil "github.com/k0sproject/k0s/inttest/common"
+)
+
+type CAPIDockerSuite struct {
+	suite.Suite
+	client           *kubernetes.Clientset
+	restConfig       *rest.Config
+	clusterYamlsPath string
+}
+
+func TestCAPIDockerSuite(t *testing.T) {
+	s := CAPIDockerSuite{}
+	suite.Run(t, &s)
+}
+
+func (s *CAPIDockerSuite) SetupSuite() {
+	kubeConfigPath := os.Getenv("KUBECONFIG")
+	s.Require().NotEmpty(kubeConfigPath)
+	// Get kube client from kubeconfig
+	restCfg, err := clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+	// t.Logf("restCfg.Host: %s", restCfg.Host)
+	// t.Logf("restCfg full:\n%+v", restCfg)
+	s.Require().NoError(err)
+	s.Require().NotNil(restCfg)
+	s.restConfig = restCfg
+
+	// Get kube client from kubeconfig
+	kubeClient, err := kubernetes.NewForConfig(restCfg)
+	s.Require().NoError(err)
+	s.Require().NotNil(kubeClient)
+	s.client = kubeClient
+
+	tmpDir := s.T().TempDir()
+	s.clusterYamlsPath = tmpDir + "/cluster.yaml"
+	s.Require().NoError(os.WriteFile(s.clusterYamlsPath, []byte(dockerClusterYaml), 0644))
+}
+
+func (s *CAPIDockerSuite) TestCAPIDocker() {
+
+	// Dump the cluster yaml to a file
+	tmpDir := s.T().TempDir()
+	tmpFile := tmpDir + "/cluster.yaml"
+	s.Require().NoError(os.WriteFile(tmpFile, []byte(dockerClusterYaml), 0644))
+
+	// Apply the child cluster objects
+	s.applyClusterObjects()
+	defer func() {
+		keep := os.Getenv("KEEP_AFTER_TEST")
+		if keep == "true" {
+			return
+		}
+		if keep == "on-failure" && s.T().Failed() {
+			return
+		}
+		s.T().Log("Deleting cluster objects")
+		s.deleteCluster()
+	}()
+	s.T().Log("cluster objects applied, waiting for cluster to be ready")
+
+	// Wait for the cluster to be ready
+	// Wait to see the CP pods ready
+	s.Require().NoError(common.WaitForStatefulSet(context.TODO(), s.client, "kmc-docker-test", "default"))
+
+	s.T().Log("Starting portforward")
+	fw, err := util.GetPortForwarder(s.restConfig, "kmc-docker-test-0", "default", 30443)
+	s.Require().NoError(err)
+
+	go fw.Start(s.Require().NoError)
+	defer fw.Close()
+
+	<-fw.ReadyChan
+
+	localPort, err := fw.LocalPort()
+	s.Require().NoError(err)
+	s.T().Log("waiting to see admin kubeconfig secret")
+	s.Require().NoError(util.WaitForSecret(context.TODO(), s.client, "docker-test-kubeconfig", "default"))
+	kmcKC, err := util.GetKMCClientSet(context.TODO(), s.client, "docker-test", "default", localPort)
+	s.Require().NoError(err)
+
+	s.T().Log("waiting for node to be ready")
+	s.Require().NoError(k0stestutil.WaitForNodeReadyStatus(context.TODO(), kmcKC, "docker-test-0", corev1.ConditionTrue))
+}
+
+func (s *CAPIDockerSuite) applyClusterObjects() {
```

Comment:
Apply done via cli as I could not get cluster api objects play nice with existing `util.CreateResources` dynamic client stuff 🤷